mapie/classification.py

```diff
@@ -663,13 +670,10 @@ def fit(
         self._check_parameters()
         cv = check_cv(self.cv)
         estimator = self._check_estimator(X, y, self.estimator)
-
         if self.image_input:
             check_input_is_image(X)
-        X, y = check_X_y(
-            X, y, force_all_finite=False, ensure_2d=self.image_input,
-            allow_nd=self.image_input, dtype=["float64", "int", "object"]
-        )
+        X, y = indexable(X, y)
+        y = _check_y(y)
```

Comment:
Are checks on `X` removed ? What about `sklearn.validation.check_array` ? It does not allow you to keep the same array type ?

Reply:
Yes, `check_array` convert `X` to numpy array, which was the very cause of the bug. `_check_y` basically ensures that `y` is 1-dimensional.

Reply:
OK, but what happens if an `X` array containing strings is provided directly to MAPIE ?

Reply:
If MAPIE is not initialized with an adequate pipeline, this will fail as soon as the `single_model_` tries to fit it. The notion of "data quality" depends on what the pipeline can handle, and MAPIE, as a general wrapper, cannot be too restrictive about it.

Reply:
OK understood, the message should be the same regardless of the use of MAPIE.